### PR TITLE
Improve miner work cache to reduce stale block

### LINKED ISSUE #651: Vaild pow no reward
Hi.
I have been testing the cluster. So I used Nanominer and Ethminer(QKC) every time miner send vaild pow to the cluster, so cluster accepted it but no reward. The last test was ten valid pows; only 7 got the reward. Is it something wrong with the cluster? Even build docker image from GitHub and tested- the same results

### PR DESCRIPTION
**Pull request by ninjaahhh**
should fix #651

### CODE REVIEW

File: quarkchain/cluster/miner.py [hunk 1]
```diff
@@ -254,8 +257,12 @@ def _mine_new_block_async(self):
 
         if now is None:  # clock open for mock
             now = time.time()
-        # 5 sec interval magic number
-        if not self.current_work or now - self.current_work.header.create_time > 5:
+        tip = self.get_header_tip_func()
+        if (
+            not self.current_work  # no cache
+            or self.current_work.header.height <= tip.height  # cache outdated
```
Comment (qizhou) [inline quarkchain/cluster/miner.py]: if tip is reset by root block, shard chain's tip may have smaller height.  I think we should check

or self.current_work.header.prev_has != tip.get_hash()

to make sure we are working on the correct tip + 1 block.

File: quarkchain/cluster/miner.py [hunk 2]
```diff
@@ -294,6 +301,13 @@ def _mine_new_block_async(self):
         # this copy is necessary since there might be multiple submissions concurrently
         block = copy.deepcopy(self.work_map[header_hash])
         header = block.header
+
+        # reject if tip updated
+        tip = self.get_header_tip_func()
+        if header.height <= tip.height:
```
Comment (qizhou) [inline quarkchain/cluster/miner.py]: Same comment as before.

File: quarkchain/cluster/miner.py (hunk 1 above)
Comment (ninjaahhh) [reply quarkchain/cluster/miner.py]: done

File: quarkchain/cluster/miner.py (hunk 2 above)
Comment (ninjaahhh) [reply quarkchain/cluster/miner.py]: done

**Comment by ninjaahhh** [comment]
> Great! May we create a taste branch and ask our community to test?

we can cherry pick this commit and make a pre-release for ppl testing

**Pull request merged**